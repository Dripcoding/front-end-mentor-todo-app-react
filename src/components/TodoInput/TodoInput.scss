@import "../../mixins";
@import "../../palette.scss";

.todoInput__container {
  @include flex(column, center, flex-start);
  background-color: $very-dark-desaturated-blue;
  height: 64px;
  width: 540px;
  padding: 20px 312px 20px 24px;
  margin-top: 48px;
  margin-bottom: 24px;

  .todoInput__input {
    @include flex(row, flex-start, center);
    height: 24px;

    input {
      @include text(#767992, 18px, normal, 400, -0.25px, 18px, left, none);
      background-color: inherit;
      border: none;
    }
  }
}