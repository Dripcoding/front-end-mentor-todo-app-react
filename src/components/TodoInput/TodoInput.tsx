import React from "react";

import TodoCheckBoxToggle from "../TodoCheckBoxToggle/index";
import "./TodoInput.scss";

const STYLE_BASE: string = "todoInput__";

const TodoInput = (): JSX.Element => {
  return (
    <div className={`${STYLE_BASE}container`}>
      <div className={`${STYLE_BASE}input`}>
        <TodoCheckBoxToggle />
        <input type="text" className={`${STYLE_BASE}todoInput`} placeholder={"Create a new todo..."} />
      </div>
    </div>
  );
};

export default TodoInput;