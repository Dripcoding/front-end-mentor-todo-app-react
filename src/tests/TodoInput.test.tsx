import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TodoInput from "../components/TodoInput/index";

describe("TodoInput", () => {
  it("matches snapshot", () => {
    const { container } = render(<TodoInput />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByTestId, getByRole, getByPlaceholderText } = render(<TodoInput />);

    const todoToggle = getByTestId("TODO_CHECKBOX_TOGGLE");
    expect(todoToggle).toBeInTheDocument();
    expect(todoToggle).toBeVisible();

    const todoInput = getByRole("textbox");
    expect(todoInput).toBeInTheDocument();
    expect(todoInput).toBeVisible();
    expect(todoInput).toHaveValue("");

    const todoInputPlaceholder = getByPlaceholderText("Create a new todo...");
    expect(todoInputPlaceholder).toBeInTheDocument();
    expect(todoInputPlaceholder).toBeVisible();
  });

  it("captures todo text correctly", () => {
    const { getByRole } = render(<TodoInput />);

    const sampleInput = "cook dinner";
    const todoInput = getByRole("textbox");
    userEvent.type(todoInput, sampleInput);

    expect(todoInput).toHaveValue(sampleInput);
  });

  it("resets when a todo is submitted", () => {
    const { getByRole } = render(<TodoInput />);

    const sampleInput = "finish homework";
    const todoInput = getByRole("textbox");
    userEvent.type(todoInput, sampleInput + "{enter}");

    expect(todoInput).toHaveValue("");
  });
});