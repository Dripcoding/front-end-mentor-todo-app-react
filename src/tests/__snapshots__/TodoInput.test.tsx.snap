// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput matches snapshot 1`] = `
<div>
  <div
    class="todoInput__container"
  >
    <div
      class="todoInput__input"
    >
      <div
        class="todoCheckBoxToggle__container"
        data-testid="TODO_CHECKBOX_TOGGLE"
      />
      <input
        class="todoInput__todoInput"
        data-testid="TODO_INPUT"
        placeholder="Create a new todo..."
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
